internal/parser/scanner/scanner.go

```diff
@@ -283,18 +284,25 @@ func (s *scanner) peekString(str string) bool {
 }
 
 // createToken creates a token with given parameters
-func createToken(line, col, start, pos int, t token.Type, value string, s *scanner) token.Token {
-	token := token.New(line, col, start, pos-start, t, value)
-	s.start = pos
-	return token
+func (s *scanner) createToken(t token.Type) token.Token {
+	tk := token.New(s.line, s.col, s.start, s.pos, t, string(s.input[s.start:s.pos]))
+	s.start = s.pos
+	return tk
 }
 
 // seekNext returns the position of the end of a keyword.
 // It takes the start position of the keyword.
 func (s *scanner) seekNext(start int) int {
-	start++
-	for s.input[start] != ' ' {
+	// start++
+	for start < len(s.input) && s.input[start] != ' ' {
```

Comment:
`' '` is not all spaces that SQL supports, thus this should be matched by a matcher (`Whitespace.Matches(s.input[start])`), and the Whitespace matcher can be extended

Reply:
Added.

Reply:
Have you committed it?